apollo-router-core/src/spec/query.rs

```diff
@@ -176,79 +194,68 @@ impl Query {
                     }
                 }
                 Selection::InlineFragment { selection_set } => {
-                    self.apply_selection_set(selection_set, input, output);
+                    self.apply_selection_set(selection_set, input, output, schema);
                 }
                 Selection::FragmentSpread { name } => {
-                    if let Some(selection_set) = self.fragments.get(name) {
-                        self.apply_selection_set(selection_set, input, output);
+                    if let Some(selection_set) = self
+                        .fragments
+                        .get(name)
+                        .or_else(|| schema.fragments.get(name))
```

Comment:
nice, that's less overhead than accumulating all of the schema's fragments at https://github.com/apollographql/router/pull/227/files#diff-3bc44ed791f17958d00dcbcdff996a241de8da3f2f5b1a3226ebf161245c4da3L50